components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,480 @@
+
+	--set resulttype to 'default' if walkover is set
+	if args.walkover then
+		args.resulttype = RESULT_TYPE_WALKOVER
+	end
+
+	local autoFinished = Logic.readBool(Logic.emptyOr(args.autofinished, true))
+	-- see if match should actually be finished if score is set
+	if isScoreSet and autoFinished and not Logic.readBool(args.finished) then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local lang = mw.getContentLanguage()
+		local matchUnixTime = tonumber(lang:formatDate('U', args.date))
+		local threshold = args.dateexact and 30800 or 86400
+		if matchUnixTime + threshold < currentUnixTime then
+			args.finished = true
+		end
+	end
+
+	-- apply placements and winner if finshed
+	local winner = tostring(args.winner or '')
+	if Logic.readBool(args.finished) then
+		args = matchFunctions.applyMatchPlacement(opponents, args)
+	-- only apply arg changes otherwise
+	else
+		for opponentIndex, opponent in pairs(opponents) do
+			args['opponent' .. opponentIndex] = opponent
+		end
+	end
+
+	-- set the match winner
+	args = matchFunctions.setMatchWinner(winner, opponents, args)
+	return args
+end
+
+function matchFunctions.getPlayers(match, opponentIndex, teamName)
+	for playerIndex = 1, MAX_NUM_PLAYERS do
+		-- parse player
+		local player = Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex]) or {}
+		player.name = player.name or Variables.varDefault(teamName .. '_p' .. playerIndex)
+		player.displayname = player.displayname
+			or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'dn', player.name:gsub('_', ' '))
+		player.flag = player.flag or Variables.varDefault(teamName .. '_p' .. playerIndex .. 'flag')
+		if not Table.isEmpty(player) then
+			match['opponent' .. opponentIndex .. '_p' .. playerIndex] = player
+		end
+	end
+	return match
+end
+
+--
+-- map related functions
+--
+function mapFunctions.getExtraData(map)
+	local overtimes = Array.extractValues(Table.mapValues(mw.text.split(map.overtime or '', ','), tonumber))
+
+	map.extradata = {
+		comment = map.comment,
+		header = map.header,
+		overtime = Table.isNotEmpty(overtimes) and Json.stringify(overtimes) or nil
+	}
+	return map
+end
+
+function mapFunctions.getScoresAndWinner(map)
+	map.scores = {}
+	local indexedScores = {}
+	for scoreIndex = 1, MAX_NUM_OPPONENTS do
+		-- read scores
+		local score = map['score' .. scoreIndex]
+		local obj = {}
+		if not Logic.isEmpty(score) then
+			if TypeUtil.isNumeric(score) then
+				score = tonumber(score)
+				obj.status = STATUS_HAS_SCORE
+				obj.score = score
+				obj.index = scoreIndex
+			elseif Table.includes(ALLOWED_STATUSES, score) then
+				obj.status = score
+				obj.score = -1
+				obj.index = scoreIndex
+			end
+			table.insert(map.scores, score)
+			indexedScores[scoreIndex] = obj
+		else
+			break
+		end
+	end
+	local isFinished = map.finished
+	if not Logic.isEmpty(isFinished) then
+		isFinished = Logic.readBool(isFinished)
+	else
+		isFinished = not Logic.readBool(map.unfinished)
```

Comment:
Is `unfinished` needed? Seems kinda redundant

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2273/commits/0d4ea2e0af67800e696695fd24221520950d0e3c